backend/workflow_manager/endpoint/destination.py

```diff
@@ -175,32 +177,41 @@ def copy_output_to_output_directory(self) -> None:
         connector: ConnectorInstance = self.endpoint.connector_instance
         connector_settings: dict[str, Any] = connector.connector_metadata
         destination_configurations: dict[str, Any] = self.endpoint.configuration
-        root_path = str(connector_settings.get(DestinationKey.PATH))
-        output_folder = str(
+        root_path = connector_settings.get(DestinationKey.PATH, "")
+
+        output_directory = str(
             destination_configurations.get(DestinationKey.OUTPUT_FOLDER, "/")
         )
-        overwrite = bool(
-            destination_configurations.get(
-                DestinationKey.OVERWRITE_OUTPUT_DOCUMENT, False
-            )
+        destination_fs = self.get_fs_connector(
+            settings=connector_settings, connector_id=connector.connector_id
         )
-        output_directory = os.path.join(root_path, output_folder)
-
+        destination_fs_cls_name = destination_fs.__class__.__name__
+        output_directory = UnstractFsConnectorHelper.get_fs_root_dir(
+            fs_cls_name=destination_fs_cls_name,
+            root_path=root_path,
+            input_dir=output_directory,
+        )
+        logger.debug(f"destination output directory {output_directory}")
         destination_volume_path = os.path.join(
             self.execution_dir, ToolExecKey.OUTPUT_DIR
         )
+        destination_fsspec = destination_fs.get_fsspec_fs()
 
-        connector_fs = self.get_fsspec(
-            settings=connector_settings, connector_id=connector.connector_id
-        )
-        if not connector_fs.isdir(output_directory):
-            connector_fs.mkdir(output_directory)
+        try:
+            is_dir = destination_fsspec.isdir(output_directory)
+            if not is_dir:
+                destination_fsspec.mkdir(output_directory)
+        except (
+            DropBoxApiError
```

Comment:
Make this Exception handler in it's own connecrot class since this exception is more related to that class only.

Reply:
@muhammad-ali-e handled in connector exception class